src/main/kotlin/fi/metatavu/megasense/dataportal/airquality/PollutantController.kt

```diff
@@ -166,31 +166,50 @@ class PollutantController {
     }
 
     /**
-     * Updates pollutant name and display name
+     * Updates pollutant intake limit and recommendation
      *
      * @param oldPollutant pollutant to update
-     * @param newPollutantName new pollutant name
-     * @param newDisplayName new display name
-     * @param newUnit new unit
      * @param newLimit intake limit amount of a pollutant
      * @param newRecommendation intake recommendation amount of a pollutant
      * @param modifierId id of user updating the value
      * @return updated pollutant
      */
     fun updatePollutant(
         oldPollutant: Pollutant,
-        newPollutantName: String,
-        newDisplayName: String,
-        newUnit: String,
         newLimit: Float,
         newRecommendation: Float,
         modifierId: UUID
     ): Pollutant {
-        val updated = pollutantDAO.updatePollutantName(oldPollutant, newPollutantName, modifierId)
-        pollutantDAO.updatePollutantDisplayName(updated, newDisplayName, modifierId)
-        pollutantDAO.updatePollutantDisplayLimit(updated, newLimit, modifierId)
+        val updated = pollutantDAO.updatePollutantLimit(oldPollutant, newLimit, modifierId)
+        return pollutantDAO.updatePollutantRecommendation(updated, newRecommendation, modifierId)
+    }
+
+    /**
+     * Updates pollutant on start
+     *
+     * @param oldPollutant pollutant to update
+     * @param newVariableName new variable name of a pollutant
+     * @param newUnit new unit of a pollutant
+     * @param newDisplayName new display name of a pollutant
+     * @param newPenaltyRequestParameter new penalty request parameter of a pollutant
+     * @param newThresholdRequestParameter new threshold request parameter a pollutant
+     * @param modifierId id of user updating the value
+     * @return updated pollutant
+     */
+    private fun updatePollutantOnStart(
```

Comment:
```suggestion
    private fun updatePollutant(
```
Consider overloading method instead of creating one with separate name?

Reply:
very good point... will do that